### PR TITLE
aws-sdk v3

### PR DESCRIPTION
**Pull request by nishanthmahasamudram**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -47,6 +47,7 @@ services:
       - AWS_EXECUTION_ENV=True
       - DEFAULT_REGION=eu-west-2
       - HOSTNAME_EXTERNAL=${HOSTNAME_EXTERNAL-localstack}
+      - SQS_PROVIDER=elasticmq
```
Comment (andrewrlee) [inline docker-compose.yml:50]: I think this probably isn't needed any more?

File: server/listeners/sqsDomainEventsListener.ts [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
 import { Services } from '../services'
 
-const sqs = new AWS.SQS({
+const sqs = new SQSClient({
   region: 'eu-west-2',
+  endpoint: config.sqs.endpoint,
+  credentials: {
+    accessKeyId: 'foo',
+    secretAccessKey: 'bar',
+  },
```
Comment (andrewrlee) [inline server/listeners/sqsDomainEventsListener.ts:14]: this shouldn't have credentials defined - these are picked up from environmental variables when running locally (from features.env in integration tests or .env when running locally), or via [IRSA](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) when running in kubernetes.

File: server/listeners/sqsPrisonEventsListener.ts [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
 import { Services } from '../services'
 
-const sqs = new AWS.SQS({
+const sqs = new SQSClient({
   region: 'eu-west-2',
+  endpoint: config.sqs.endpoint,
+  credentials: {
+    accessKeyId: 'foo',
+    secretAccessKey: 'bar',
+  },
```
Comment (andrewrlee) [inline server/listeners/sqsPrisonEventsListener.ts:14]: same here

File: server/listeners/sqsProbationEventsListener.ts [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
 import { Services } from '../services'
 
-const sqs = new AWS.SQS({
+const sqs = new SQSClient({
   region: 'eu-west-2',
+  endpoint: config.sqs.endpoint,
+  credentials: {
+    accessKeyId: 'foo',
+    secretAccessKey: 'bar',
+  },
```
Comment (andrewrlee) [inline server/listeners/sqsProbationEventsListener.ts:14]: and here

File: integration_tests/support/events.ts [hunk 5]
```diff
@@ -1,13 +1,18 @@
...
+const { SQS } = AWS_SQS
+
+const sqs = new SQS({
   region: 'eu-west-2',
-  accessKeyId: 'foo',
-  secretAccessKey: 'bar',
+  credentials: {
+    accessKeyId: 'foo',
+    secretAccessKey: 'bar',
+  },
```
Comment (andrewrlee) [inline integration_tests/support/events.ts:11]: this credentials shouldn't be set as its picked up from features.env

File: bin/inbound-message-prisoner-released.sh [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+read -r -p "Please enter the prison code: " PRISONER_CODE
+
+if [ -z "$PRISONER_CODE" ]
+then
+  echo "No prison code specified."
+  exit 99
+fi
+
+aws --region=eu-west-2 --endpoint-url=http://localhost:4566 sns publish --topic-arn arn:aws:sns:eu-west-2:000000000000:domainevents-topic --message '{"eventType":"prison-offender-events.prisoner.released","version":"1.0","occurredAt":"2020-02-12T15:14:24.125533+00:00", "publishedAt":"2020-02-12T15:15:09.902048716+00:00","description":"A prisoner has been released from prison", "additionalInformation":{"nomsNumber":"'"$PRISONER_NUMBER"'", "prisonId":"'"$PRISONER_CODE"'","reason":"RELEASED", "probableCause":"RECALL", "source":"PROBATION", "details":"Recall referral date 2021-05-12"}}' --message-attributes '{"eventType":{"DataType":"String","StringValue":"prison-offender-events.prisoner.released"}}'
```
Comment (andrewrlee) [inline bin/inbound-message-prisoner-released.sh:19]: meh.. missing new line at end of file

File: docker-compose.yml (hunk 1 above)
Comment (nishanthmahasamudram) [reply docker-compose.yml:50]: Yes, you are right.
I will remove this flag.

File: integration_tests/support/events.ts (hunk 5 above)
Comment (nishanthmahasamudram) [reply integration_tests/support/events.ts:11]: No, It's not picking up credentials from feature.env file.

Comment (andrewrlee) [reply integration_tests/support/events.ts:11]: my bad! this runs from within cypress, feature.env is only available to the running application!

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**